artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -11,13 +12,17 @@ pub fn now(interp: &mut Artichoke) -> Result<Value, Error> {
     Ok(result)
 }
 
-pub fn at<T>(interp: &mut Artichoke, args: T) -> Result<Value, Error>
-where
-    T: IntoIterator<Item = Value>,
-{
-    let _ = interp;
-    let _ = args;
-    Err(NotImplementedError::new().into())
+pub fn at(interp: &mut Artichoke, seconds: Value, microseconds: Value) -> Result<Value, Error> {
+    let seconds = implicitly_convert_to_int(interp, seconds)?;
+    let sub_second_nanos = if microseconds.is_nil() {
+        0
+    } else {
+        implicitly_convert_to_int(interp, microseconds)? * 1_000
+    };
```

Comment:
It looks like MRI adds overflow microseconds to the `seconds` parameter, which means I think we'll want something like:

```suggestion
    let seconds = implicitly_convert_to_int(interp, seconds)?;
    let (seconds, sub_second_nanos) = if let Some(microseconds) = microseconds {
        let microseconds = implicitly_convert_to_int(interp, microseconds)?;
        let overflow_seconds = microseconds / 1_000_000;
        let sub_second_nanos = u32::try_from((microseconds % 1_000_000) * 1_000).map_err(|_| ArgumentError::with_message("Time too large"));
        let seconds = seconds.checked_add(overflow_seconds).ok_or(|| ArgumentError::with_message("Time too large"))?;
        (seconds, sub_second_nanos)
    } else {
        (seconds, 0_u32)
    };
```

This setup ensures we return an error if we overflow, ensures `sub_second_nanos` is in an acceptable range, and allows us to avoid an `as` cast which might truncate or otherwise lose precision.

Reply:
After reviewing the `Time::at` impl in `spinoso-time`, I like the direction you've taken to accept `i64` in `spinoso-time`, so the suggestions above are best relocated to `Time::at` in `spinoso-time`. Here' all we should be doing is unmarshalling arguments, so this suggestion should probably look like:

```suggestion
    let seconds = implicitly_convert_to_int(interp, seconds)?;
    let sub_second_nanos = if let Some(microseconds) = microseconds {
        implicitly_convert_to_int(interp, microseconds)?.checked_mul(1_000).ok_or_else(|| ArgumentError::with_message("Time too large"))?;
    } else {
        0_i64
    };
```

Can we turn `1_000` into a named constant in `spinoso-time` that we use here?